adds faction modifier to mob spawner

MuckerMayhem opened the pull request:
:cl: Mucker
admin: Adds a faction modifier to mob spawner buildmode.
/:cl:

also does minor cleanup

SierraKomodo commented on code/modules/admin/buildmode/mob_spawning.dm:
```diff
@@ -107,71 +108,56 @@ GLOBAL_LIST_INIT(mob_spawners, list())
-			return TOPIC_HANDLED
+		if (matches.len)
+			var/mob/living/simple_animal/chosen
+			if (matches.len==1)
+				chosen = matches[1]
+			else
+				chosen = input("Choose a mob", "Mob") as null | anything in matches
```
Byond automatically selects the only option if there's only one option in the input, if I remember correctly. (I learned this one working on teleporter code)

```suggestion
			chosen = input("Choose a mob", "Mob") as null | anything in matches
```

SierraKomodo commented on code/modules/admin/buildmode/mob_spawning.dm:
```diff
@@ -107,71 +108,56 @@ GLOBAL_LIST_INIT(mob_spawners, list())
-
-		if (!(chosen in spawner.mobs))
-			spawner.mobs += chosen
+			if (chosen && !(chosen in spawner.mobs))
+				spawner.mobs += chosen
```
`|=` checks for existing entries before adding a new one.

```suggestion
			if (chosen)
				spawner.mobs |= chosen
```

SierraKomodo commented on code/modules/admin/buildmode/mob_spawning.dm:
```diff
@@ -180,23 +166,29 @@ GLOBAL_LIST_INIT(mob_spawners, list())
+	if (href_list["set_faction"])
+		var/faction = input("Enter a faction name, leave blank to set it to default", "Mob Spawner") as text | null
+
+		if (faction == "")
+			faction = null
+
+		spawner.faction = faction
```
`null` and `""` are not considered equivalent, so this should safely be able to handle null values (cancel) as canceling the prompt.

Honestly though, just adding an `X`, clear, or delete button to the field would be more intuitive for the UI.

```suggestion
		if (!isnull(faction)
			if (faction == "")
		  		faction = null

			spawner.faction = faction
```

MuckerMayhem committed:
adds faction modifier to mob spawner

Adds faction modifier to mob spawner, and also cleans up some unnecessary redundancies.
